src/webapp/components/form/FormState.ts

```diff
@@ -133,3 +302,82 @@ export function updateFieldState<F extends FormFieldState>(
 ): F {
     return { ...fieldToUpdate, value: newValue };
 }
+
+// VALIDATIONS:
+
+export function isValidForm(formSections: FormSectionState[]): boolean {
+    const allFields: FormFieldState[] = getAllFieldsFromSections(formSections);
+
+    return allFields.every(field => {
+        const validationErrors = validateField(field, allFields);
+
+        return !validationErrors || validationErrors.errors.length === 0;
+    });
+}
+
+export function validateForm(
+    formState: FormState,
+    updatedField: FormFieldState
+): ValidationError[] {
+    return validateSections(formState.sections, updatedField, formState);
+}
+
+function validateSections(
```

Comment:
can we move all section related functions to seperate file called FormSections or something like that for better understandability?

Reply:
I have created also another file with field related functions